src/poetry/console/commands/init.py

```diff
@@ -262,28 +285,13 @@ def _determine_requirements(
                     package = self.ask("\nAdd a package:")
                     continue
 
-                matches = self._get_pool().search(constraint["name"])
-
+                canonicalized_name = canonicalize_name(constraint["name"])
+                matches = self._get_pool().search(canonicalized_name)
                 if not matches:
                     self.line("<error>Unable to find package</error>")
                     package = False
                 else:
-                    choices = []
-                    matches_names = [p.name for p in matches]
-                    exact_match = constraint["name"] in matches_names
-                    if exact_match:
-                        choices.append(
-                            matches[matches_names.index(constraint["name"])].pretty_name
-                        )
-
-                    for found_package in matches:
-                        if len(choices) >= 10:
-                            break
-
-                        if found_package.name.lower() == constraint["name"].lower():
-                            continue
-
-                        choices.append(found_package.pretty_name)
+                    choices = self._generate_choice_list(matches, canonicalized_name)
 
                     self.line(
```

Comment:
While we are here. Can you extend this by an info in the case the list is truncated? Then we can close https://github.com/python-poetry/poetry/issues/5033 as well.